bpf/lib/encrypt.h

```diff
@@ -68,7 +68,11 @@ do_decrypt(struct __ctx_buff *ctx, __u16 proto)
 		return CTX_ACT_OK;
 	}
 	ctx->mark = 0;
+#ifdef ENABLE_ROUTING
+	return CTX_ACT_OK;
+#else
 	return redirect(CILIUM_IFINDEX, 0);
+#endif /* ENABLE_ROUTING */
```

Comment:
This might work but it seems in conflict with the documented behaviour of the routing flag in userspace:

```
$ git grep -A 2 RequireRouting pkg/datapath/
pkg/datapath/config.go: // RequireRouting returns true if the endpoint requires BPF routing to
pkg/datapath/config.go- // be enabled, when disabled, routing is delegated to Linux routing
pkg/datapath/config.go: RequireRouting() bool
pkg/datapath/config.go-
pkg/datapath/config.go- // RequireEndpointRoute returns true if the endpoint wishes to have a
--
pkg/datapath/linux/config/config.go:    if e.RequireRouting() {
pkg/datapath/linux/config/config.go-            fmt.Fprintf(fw, "#define ENABLE_ROUTING 1\n")
pkg/datapath/linux/config/config.go-    }
```

Reply:
Agree its a bit of a conflict and might be fragile/broke. I guess the right thing to do is push RequireEndpointRoute into a define and use that.

Reply:
Updated to use a new endpoint routes enabled define.